### PR TITLE
Fixes and enhancement in solver exploration

### PR DESCRIPTION
**Pull request by Necktschnagge**
Fixes:
* exploration on initial state being final (reported from GUI usage, fixes: #107)
* check for final state in case of multiple target pieces using color-agnostic mode (found by static code analysis, never practically reproduced this bug)

### CODE REVIEW

**Comment by Necktschnagge** [comment]
Need to add a test for solver run on initial state:

I did not yet check what happens with optimal move paths.
Such a path is an empty word. We should check if this is handled correctly.


OK, wait, anecdotal evidence found here: https://github.com/Necktschnagge/tobor-games/issues/107#issuecomment-1902715759

The ListView displays exaclty one optimal path consisting of the empty word.
The other ones are displayed since these represent final states that had been reached through exploration.

**Comment by Necktschnagge** [comment]
I tested the behavour of optimal empty path locally by hand.
It works like in the picture mentioned above.

Further optimization can be made during static code analysis.
The engine works fine here.

**Comment by Necktschnagge** [comment]
### DevOps

* Prepare a test (manual in GUI and as part of google test) to run solver on specific boards / initial states.
* Better output of current bord / state ids would be a plus.
* Get board + state by entering the ids.

**Review by r0ost3r — APPROVED** [review]

File: src/models.h [hunk 1]
```diff
@@ -509,6 +509,10 @@ namespace tobor {
 
 			// cell_id_type::int_type for cell ids
 
+
+			// here it should not use a dependent name without alias-defining it in the class itself!
```
Comment (Necktschnagge) [inline src/models.h:513]: todo later

Comment (Necktschnagge) [reply src/models.h:513]: subsumed in #98

**Pull request merged**